### PR TITLE
HTMLExporter: Add embed_images option for embedding images from Markdown cells as base64

### PR DESCRIPTION
**Pull request by martinRenou**

### CODE REVIEW

File: nbconvert/filters/markdown_mistune.py [hunk 1]
```diff
@@ -28,6 +31,10 @@
 from nbconvert.filters.strings import add_anchor
 
 
+class InvalidNotebook(Exception):
```
Comment (martinRenou) [inline nbconvert/filters/markdown_mistune.py:34]: This un unrelated to the PR itself, `InvalidNotebook` was used in this file but never imported. Importing it was resulting in a circular import so we need to define it here (it's only used in this file anyway)

File: nbconvert/nbconvertapp.py [hunk 2]
```diff
@@ -56,6 +56,7 @@ def validate(self, obj, value):
     'template' : 'TemplateExporter.template_name',
     'template-file' : 'TemplateExporter.template_file',
     'theme' : 'HTMLExporter.theme',
+    'embed-images' : 'HTMLExporter.embed_images',
```
Comment (bollwyvl) [inline nbconvert/nbconvertapp.py:59]: perhaps make it a `flag`, if it defaults to `False`?

Comment (martinRenou) [reply nbconvert/nbconvertapp.py]: thanks for the review! Makes total sense, I did the change

**Comment by maartenbreddels** [comment]
Related to https://github.com/voila-dashboards/voila/pull/101

**Pull request merged**